src/main/java/com/epam/team1/emotional/help/service/UserService.java

```diff
@@ -1,11 +1,71 @@
 package com.epam.team1.emotional.help.service;
 
+import com.epam.team1.emotional.help.dto.UserAddDataRequestDto;
+import com.epam.team1.emotional.help.dto.UserResponseDTO;
+import com.epam.team1.emotional.help.exception.WrongUserIdException;
+import com.epam.team1.emotional.help.mappers.UserMapper;
+import com.epam.team1.emotional.help.model.User;
+import com.epam.team1.emotional.help.repository.UserRepository;
+import com.epam.team1.emotional.help.security.UserDetailsImplementation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
+import javax.persistence.EntityNotFoundException;
+
 @Service
+@RequiredArgsConstructor
 public class UserService {
-    public Boolean userIsAuthorized(){
+
+    private final UserRepository userRepository;
+
+    private final UserMapper userMapper;
+
+    public Boolean userIsAuthorized() {
         //TODO finish it
         return Boolean.TRUE;
     }
+
+    public User getCurrentUser() {
+        User currentUser;
+        try {
+            currentUser = ((UserDetailsImplementation) SecurityContextHolder.
+                    getContext().
+                    getAuthentication().
+                    getPrincipal()).getUser();
+        } catch (NullPointerException e) {
+            return null;
+        }
+        return currentUser;
```

Comment:
Looks really strange

Reply:
can I optimize this code in the next branch